src/main/java/software/bernie/techarium/tile/voltaicpile/VoltaicPileTile.java

```diff
@@ -0,0 +1,81 @@
+package software.bernie.techarium.tile.voltaicpile;
+
+import net.minecraft.block.FireBlock;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.Explosion;
+import software.bernie.techarium.block.voltaicpile.VoltaicPileBlock;
+import software.bernie.techarium.machine.sideness.FaceConfig;
+import software.bernie.techarium.machine.sideness.Side;
+import software.bernie.techarium.tile.base.TechariumTileBase;
+import software.bernie.techarium.trait.tile.TileBehaviours;
+
+import static software.bernie.techarium.registry.BlockRegistry.VOLTAIC_PILE;
+
+public class VoltaicPileTile extends TechariumTileBase {
+    public VoltaicPileTile() {
+        super(VOLTAIC_PILE.getTileEntityType(), TileBehaviours.voltaicPile);
+
+        for (Side side: Side.values()) {
+            getFaceConfigs().put(side, FaceConfig.PUSH_ONLY);
+        }
+    }
+
+    @Override
+    public ActionResultType onTileActivated(PlayerEntity player) {
+        return ActionResultType.PASS;
+    }
+
+    @Override
+    public void tick() {
+        getPowerTrait().ifPresent(trait -> {
+            // Prevent immediate explosion of the Voltaic Pile, the getDayTime is a misleading name, it also includes the night
+            if (level.getDayTime() % 40 == 0 && !level.isClientSide()) {
+                for (Direction d : Direction.values()) {
+                    if (level.getBlockState(worldPosition.relative(d)).getBlock() instanceof FireBlock && trait.getEnergyStorage().getEnergyStored() > 0) {
+                        explode();
+                    }
+                }
```

Comment:
add the following to push power like it's common in generators and other blocks`
if (!level.isClientSide()) {
    for (Direction direction : Direction.values()) {
        TileEntity te = level.getBlockEntity(worldPosition.relative(direction));
        if (te != null) {
            LazyOptional<IEnergyStorage> lazyOtherEnergy = te.getCapability(CapabilityEnergy.ENERGY, direction.getOpposite());
            lazyOtherEnergy.ifPresent(otherEnergy -> {
                int extractable = trait.getEnergyStorage().extractEnergy(Integer.MAX_VALUE, true);
                int received = otherEnergy.receiveEnergy(extractable, false);
                trait.getEnergyStorage().extractEnergy(received, false);
            });
        }
    }
}
`

Reply:
i already implemented a way for it to push power

Reply:
I couldn't find it in code and when i placed the pile next to a botarium it didn't push energy, but I'll tesy it again

Reply:
For some reason it didn't work when I tried to use the voltaic pile, but probably was just because the botarium didn't accept energy form the front... my bad